opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -30,24 +31,43 @@ class Aggregator(abc.ABC):
     """
 
     def __init__(self, config=None):
-        self.current = None
-        self.checkpoint = None
-        if config:
+        self._lock = threading.Lock()
```

Comment:
Technically, all classes that extend from this would want this implementation, but design-wise, what are we deciding for having implementation within ABCs? This applies to `merge` and `verify_type` as well. See [this comment](https://github.com/open-telemetry/opentelemetry-python/pull/341#discussion_r369310574).

Reply:
I think it is the right thing to have implementation in the base class if that implementation can be used by the concrete classes. There are 2 different things happening here, first, `abstractmethod` makes it mandatory for the concrete class to implement a specific method. Second, having implementation in the parent class. The second thing does not go against the first one, it is independent and it fulfills a separate need, the one of having common code in one single place. We can move the lock instantiation to the concrete classes if that is the project consensus, but I think that would just mean more repeated code :man_shrugging:

Reply:
Yeah personally I don't mind having implementation in the base class. Just something I'd thought I'd bring up.